src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java

```diff
@@ -416,7 +417,9 @@ private boolean isPlayerAttacking()
         Actor target = client.getLocalPlayer().getInteracting();
         if (target != null && (target instanceof NPC))
         {
-            boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0;
+            // Turns out fishing spots look like target dummies... so check it's not one
+            boolean isNotFishingSpot = getIsNotFishingSpot(target);
+            boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0 && isNotFishingSpot;
```

Comment:
Is there a list of things that have a health ratio? I think this is a potential source of false postivies.

 I've noticed false positives on the attack countdown when interacting with non-attackable NPCs, usually while on a casting cooldown. For example, I was on high alch cooldown and the attack timer triggered when I got a new Mahogany Homes contract from Amy.

Reply:
Granted, false positives are preferable to false negatives if we have to make a tradeoff.

Reply:
> For example, I was on high alch cooldown and the attack timer triggered when I got a new Mahogany Homes contract from Amy.

Oh wow thats quite unfortunate. Yeah I can easily reproduce that by just walking towards an NPC with any non-zero animation. We could take inspiration from the xp tracker plugin which needs to find exp modifiers:

```java
if (interacting instanceof NPC && COMBAT.contains(skill))
{
	final int xpModifier = worldSetToType(client.getWorldType()).modifier(client);
	final NPC npc = (NPC) interacting;
	xpState.updateNpcExperience(skill, npc, npcManager.getHealth(npc.getId()), xpModifier);
}
```

Here the `getHealth()` might be more reliable that health ratio?

There's also an option to explore which is looking for an attack option in the menu options of a given NPC. I'll have a play with both...

Reply:
So using the menu and hard coding the dummy_id works much better defensively reducing the false positives while not affecting the false negatives. I'm going to test it a bit more before committing it and pushing. 

This can still be triggered by alching and interacting with an NPC which is attackable, but this would fix the fishing spots, Amy and that class of interacting issues (some pick-pocketing, etc) with NPCs which don't have health and levels. 

It still leaves me unsatisfied because pick-pocketing knights, etc will still be an "attack". We could go with a block list here? A list of animations we know which aren't attacks for the rarer NPCs which have attack options but can also be interacted with which isn't an attack.


``` diff
diff --git a/src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java b/src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
index 853046a..0855c03 100644
--- a/src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
+++ b/src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java
@@ -32,2 +32,4 @@ import java.awt.Color;
 import java.awt.Dimension;
+import java.util.Arrays;
+
 import javax.inject.Inject;
@@ -59,2 +61,3 @@ import net.runelite.client.game.FishingSpot;
 import net.runelite.client.game.ItemManager;
+import net.runelite.client.game.NPCManager;
 import net.runelite.client.plugins.Plugin;
@@ -102,2 +105,5 @@ public class AttackTimerMetronomePlugin extends Plugin
 
+    @Inject
+	private NPCManager npcManager;
+
     public int tickPeriod = 0;
@@ -410,2 +416,4 @@ public class AttackTimerMetronomePlugin extends Plugin
 
+    private static final int TARGET_DUMMY_ID = 10507;
+
     private boolean isPlayerAttacking()
@@ -419,6 +427,7 @@ public class AttackTimerMetronomePlugin extends Plugin
         {
-            // Turns out fishing spots look like target dummies... so check it's not one
-            boolean isNotFishingSpot = getIsNotFishingSpot(target);
-            boolean isTargetDummy = target.getHealthRatio() == -1 && target.getCombatLevel() == 0 && isNotFishingSpot;
-            boolean attackingNPC = target.getCombatLevel() > 0 || isTargetDummy;
+            final NPC npc = (NPC) target;
+            boolean containsAttackOption = Arrays.stream(npc.getComposition().getActions()).anyMatch("Attack"::equals);
+            Integer health = npcManager.getHealth(npc.getId());
+            boolean hasHealthAndLevel = health != null && health > 0 && target.getCombatLevel() > 0;
+            boolean attackingNPC = hasHealthAndLevel || npc.getId() == TARGET_DUMMY_ID || containsAttackOption;
             // Not walking is either any player animation or the edge cases which don't trigger an animation, e.g Salamander.
@@ -441,6 +450,5 @@ public class AttackTimerMetronomePlugin extends Plugin
 
-    private boolean getIsNotFishingSpot(Actor target) {
-        final NPC npc = (NPC) target;
+    private boolean getFishingSpot(NPC npc) {
         FishingSpot spot = FishingSpot.findSpot(npc.getId());
-        return spot == null;
+        return spot != null;
     }


```

Reply:
2t skilling methods are also affected by this issue but atleast that one sort of makes sense 😅

Reply:
[03f6a3a](https://github.com/ngraves95/attacktimer/pull/56/commits/03f6a3a3e7701e21be43117c263f27f3d85c9887) is my best attempt at reducing false positives, it introduces a block list of animations (only pick-pocket on that list for now). 

Happy to revert the block list if you think that direction is poor. There's still ways to trigger the timer incorrectly, like being hit (flinching animation) then interacting, or alching while interacting with an NPC with a health bar (talking, still walking to attack the NPC). Having no ammo in the quiver.

Reply:
Testing it also fixes https://github.com/ngraves95/attacktimer/issues/55 though off the back of this testing I've added flinching to the exclude list because that frequently mis-triggers it. 4c0c29f

Just say if you want me to rebase/squash to one commit.